### PR TITLE
[wip] Install phases termination

### PR DESCRIPTION
**Pull request by mjudeikis**
Terminate work and end lease after each install phase

### CODE REVIEW

**Comment by mjudeikis** [comment]
I dont like this enough to merge. Need to think how to sort out some of our backend->install transitions to make it more readable and manageable. Current solution (to pass in `stop` channel looks like a hack). But we have 2 layers of independent stuff in the middle :/

File: pkg/api/openshiftcluster.go [hunk 1]
```diff
@@ -204,4 +204,5 @@ const (
 	InstallPhaseDeployStorage InstallPhase = iota
 	InstallPhaseDeployResources
 	InstallPhaseRemoveBootstrap
+	InstallPhaseConfigureCluster
```
Comment (m1kola) [inline pkg/api/openshiftcluster.go]: FYI https://github.com/Azure/ARO-RP/issues/106

Comment (mjudeikis) [reply pkg/api/openshiftcluster.go]: I think we shoud have more, not less. Just because they take ages and it easier to handle restarts

Comment (asalkeld) [reply pkg/api/openshiftcluster.go]: I think @jim-minter was of the opinion of less is better. I am in the process of refactoring this code to have 2 steps.

**Pull request closed without merging**